packages/turf-shortest-path/index.js

```diff
@@ -0,0 +1,125 @@
+var helpers = require('@turf/helpers');
+var pointGrid = require('@turf/point-grid');
+var distance = require('@turf/distance');
+var bbox = require('@turf/bbox');
+var clone = require('@turf/clone');
+var scale = require('@turf/transform-scale');
+var cleanCoords = require('@turf/clean-coords');
+var bboxPolygon = require('@turf/bbox-polygon');
+var invariant = require('@turf/invariant');
+var getType = invariant.getType;
+var lineString = helpers.lineString;
+var inside = require('@turf/inside');
+var isNumber = helpers.isNumber;
+var getCoord = invariant.getCoord;
+var gridToMatrix = require('grid-to-matrix');
+
+var jsastar = require('javascript-astar');
+var Graph = jsastar.Graph;
+var astar = jsastar.astar;
+
+// var ezStar = require('easystarjs');
+// var pathfinding = require('pathfinding');
+
+/**
+ * Returns the shortest {@link LineString|path} from {@link Point|start} to {@link Point|end} without colliding with
+ * any {@link Feature} in {@link FeatureCollection<Polygon>| obstacles}
+ *
+ * @name shortestPath
+ * @param {Geometry|Feature<Point>} start point
+ * @param {Geometry|Feature<Point>} end point
+ * @param {GeometryCollection|FeatureCollection<Polygon>} obstacles polygons
+ * @param {Object} [options={}] optional parameters
+ * @param {string} [options.units="kilometers"] can be degrees, radians, miles, kilometers, ...
+ * @param {number} [options.resolution=100] distance between matrix points on which the path will be calculated
+ * @returns {Feature<LineString>} shortest path between start and end
+ * @example
+ * var start = turf.point([-5, -6]);
+ * var end = turf.point([9, -6]);
+ * var obstacles = turf.featureCollection([turf.polygon([[0,-7],[5,-7],[5,-3],[0,-3],[0,-7]]));
+ *
+ * var path = shortestPath(start, end, obstacles);
+ *
+ * //addToMap
+ * var addToMap = [start, end, obstacles, path];
+ */
+module.exports = function (start, end, obstacles, options) {
+    // validation
+    if (!start) throw new Error('start point is required');
+    if (getType(start) !== 'Point') throw new Error('start must be Point');
+    if (!end) throw new Error('end point is required');
+    if (getType(end) !== 'Point') throw new Error('end must be Point');
+    if (!obstacles) throw new Error('obstacles collection is required');
```

Comment:
Could we possibly have obstacles as an optional parameter? In case you want to use this module over an iterative process and maybe in some scenarios you would have 2 points with or without an obstacle, we could simply just return a lineString from both points (same as an empty `featureCollection`).

Reply:
Yes! Great idea 👍